### ISSUE TITLE
Command help

### ISSUE THREAD

**Issue by thethomasbeck** [pachanka/Ptty#9]
Not really an issue, more of if you could provide an additional example for a couple use-cases - 

I have been trying to figure out a way to configure variables from input, after the command is issued. In my case, I am trying to issue a command to configure, but configure being an option of the command.

command configure

I have tried if(cmd.hasOwnProperty('configure')) and I have also tried it with next : 'command %cmd%' as well as using - 

console.log($ptty.tokenize(
    "command namespace action",
    [1, 2, 3]
));
// { 1: "namespace", 2: "action" }

data : { VARIABLE : %cmd% }

Of course command is referencing my specific command name, but I Am trying to setup multiple options with the command that allows 3 levels deep, for example - 

command <namespace> <action> (stuff specific to the action, like SOURCE/DESTINATION, etc) <options>

I can get the command to work easily, but adding the option outputs the prompt I need, but I cannot get it save. It either quits the prompt, tries to use next to add to the command and run it, or i do not get any output at all. 

TL;DR Are you able to assist with a couple more examples, specifically - 

1. command configure
(This will prompt the user for fields to pass, i.e. Name, Email, Field3, Field4, etc, I would like to save each field into a specific variable which I can later reference in an output)

Harder example I am going for :)

2. command <namespace> <action> SOURCE DESTINATION <options>
(Technically the <action> in this case is the actual command, I just want to precede it with a namespace which has to be referenced, kind of like --

console.log($ptty.tokenize(
    "python script.py -xyz --foo 'something' 'something' ",
    ['-xyz', '--foo']
));
// { -xyz: false, --foo: ""something" "something" }Ex.Ex

Ex.  command action -xyz --foo 'something' 'something'

**Comment by pachanka** [comment]
Hi @thethomasbeck 

You were on the right track using the next `$ptty.set_command_option()` object property. Here are some example commands to get you towards where you're going.

1. For your configuration command, I've composed an [explanation script](https://gist.github.com/pachanka/82cb4d783abcac61ba9897290a0a3941) that stores variables in a callbefore (a thing called before an ajax call command) and later displays the information saved. You can test it [here]().

2. If I understand correctly you want to have a command that grabs 2 arguments, like:

```
$ mv source/dir/file destination/dir/file
```

How you go about this really depends on what the arguments are going to contain because the tokenizer method requires white spaces to be enclosed in single or double quotes, which can get tricky.

If you just want any old string its not complicated at all:

```js
$ptty.register('command', {
    name : 'salute',
    method : function(cmd){
        // Get input
        var input = $.trim($ptty.get_command_option('last'));
        // Convert to array and remove command name
        input = input.split( /\s+/ );
        input.shift();
        // Convert to Title Case
        for (var i = input.length - 1; i >= 0; i--) {
            input[i] = input[i].charAt(0).toUpperCase()+input[i].substr(1).toLowerCase();
        }

        // Handle output
        if(!input.length){
            cmd.out = 'Usage: salute [first] [middle] [last name]' ;
        }else{
            if(input.length == 1){
                cmd.out = '<b>Hi '+input[0]+'! Hows it going?</b>';
            }else if(input.length == 2){
                cmd.out = '<b>The names is '+input[1]+', '+input[1]+' '+input[0]+'.</b>';
            }else{
                cmd.out = '<b>Hello '+input.join(' ')+'!</b>';
            }
        }

        return cmd;
    },
    help: 'Salutes someone who inputs their name.'
});
```
As you can see the first part of the command deals with formatting, and the second deals with the output, so the last name will always be at `input[1]` and the persons given name will always be `input[0]`.

Does that work for you, or did I miss something?

**Comment by thethomasbeck** [comment]
I think that helps explain it. At first, when I was seeing last, next, etc I was reading it as it would append the input to the command, so it would run the command with the input, but I see now what you mean, how using the first, last, split etc, you can essentially strip out the specific input you need and piece it together. I think I am starting to get it :) 

Gonna try it out and I will message back if any issues, but thank you very much for the quick reply.

**Comment by thethomasbeck** [comment]
Looks like it is working great, This should give a great start. I am trying to build a tutorial for using CLI commands. I want to help get them comfortable with the commands, configuring, running basic functions, etc before installing locally on their computer. I like that it is completely HTML/CSS/JS. This allow me to host it serverless with added security :)

Thanks again for the quick response

**Comment by pachanka** [comment]
Love that you like it.  Let me know if you run into any other speed bumps.
